week2/solutions/ccbill_project/pay_one_year_bisection_search.py

```diff
@@ -54,26 +54,26 @@ def pay_one_year_bi(balance, annual_interest_rate):
 
 def get_fixed_rate(balance, monthly_interest, upper_bound, lower_bound):
   monthly_payment = round(lower_bound + (upper_bound - lower_bound)/2, 4)
-  remain_balance = check_monthly_payment(balance, monthly_interest, monthly_payment)
+  remaining_balance = check_monthly_payment(balance, monthly_interest, monthly_payment)
 
-  if(remain_balance <= 0 and remain_balance >= -0.01):
+  if -0.01 <= remaining_balance and remaining_balance <= 0:
     return monthly_payment
-  elif remain_balance < -0.01:
+  elif remaining_balance < -0.01:
     upper_bound = monthly_payment
     monthly_payment = round(lower_bound + (upper_bound - lower_bound)/2, 4)
-  elif remain_balance > 0: 
+  elif remaining_balance > 0: 
     lower_bound = monthly_payment
     monthly_payment = round(lower_bound + (upper_bound - lower_bound)/2, 4)
   return get_fixed_rate(balance, monthly_interest, upper_bound, lower_bound)
 
 def check_monthly_payment(balance, monthly_interest, monthly_payment):
-  remain_balance = balance
+  remaining_balance = balance
```

Comment:
Lines 63 and 66 are the same. You can simplify that to one line by moving it just before the return.